### ISSUE TITLE
Massive code duplication in tests

### ISSUE THREAD

**Issue by vasily-kirichenko** [Spreads/Spreads#16]
Like this:
```fsharp
let ok, value = rc.TryGetValue(1)
ok |> should equal true
value |> should equal 2

let ok, value = rc.TryGetValue(3)
ok |> should equal true
value |> should equal 4

let ok, value = rc.TryGetValue(5)
ok |> should equal true
value |> should equal 6

let ok, value = rc.TryGetValue(7)
ok |> should equal true
value |> should equal 8

let ok, value = rc.TryGetValue(10)
ok |> should equal false
```
or like this:
```fsharp
let map = SortedMap<int, int>()
map.Add(1,1)
map.Add(3,3)
map.Add(5,5)
map.Add(7,7)

let expected = SortedMap<int, int>()
expected.Add(1,1)
expected.Add(3,4)
expected.Add(5,9)
expected.Add(7,16)
```
This:
```fsharp
let ok, value = rc.TryGetValue(2)
ok |> should equal true
(repeated :> IReadOnlyOrderedMap<_,_>).TryGetValue(2) |> snd |> should equal 1
value |> should equal 1

let ok, value = rc.TryGetValue(4)
ok |> should equal true
(repeated :> IReadOnlyOrderedMap<_,_>).TryGetValue(4) |> snd |> should equal 3
value |> should equal 3

let ok, value = rc.TryGetValue(6)
ok |> should equal true
(repeated :> IReadOnlyOrderedMap<_,_>).TryGetValue(6) |> snd |> should equal 5
value |> should equal 5

let ok, value = rc.TryGetValue(123)
ok |> should equal true
(repeated :> IReadOnlyOrderedMap<_,_>).TryGetValue(123) |> snd |> should equal 7
value |> should equal 7
```
And this:
```fsharp
for r in 0..0 do
  let mutable res = Unchecked.defaultof<_>
  perf (count/100L) "DeedleSeries Operator Add/divide Chained" (fun _ ->
    res <- ((!deedleSeries + 123456.0) /789.0) * 10.0
    ()
  )
  if res.ValueCount < 1 then failwith "avoid gc" 
  ()

for r in 0..0 do
  let mutable res = Unchecked.defaultof<_>
  perf (count/100L) "DeedleSeries Add/divide Inline" (fun _ ->
    res <- !deedleSeries 
      |> Series.map (fun _ x -> ((x + 123456.0)/789.0)*10.0) 
    ()
  )
  if res.ValueCount < 1 then failwith "avoid gc" 
  ()

for r in 0..0 do
  let mutable res = Unchecked.defaultof<_>
  perf (count/100L) "DeedleSeries Operator Add/divide" (fun _ ->
    res <- ((!deedleSeries + 123456.0)/789.0) * 10.0
    ()
  )
  if res.ValueCount < 1 then failwith "avoid gc" 
  ()
```

**Comment by buybackoff** [comment]
I used to look into MOps numbers prints-outs, tests were often copy-pasted and beauty was not a goal. Lack of proper testing suite is mentioned in #8 and Readme. You are welcome to change whatever you do not like or miss here :)

I have started [this file](https://github.com/Spreads/Spreads/blob/master/tests/Spreads.Collections.Tests/Contracts/ContractsTests.cs). The idea is to feed into it a lazy series and materialized series with correct numbers and then assert that calculations are correct.

**Comment by buybackoff** [comment]
ZipN #1 is much much more important, it is the most important piece and I know that it doesn't work correctly on some edge cases.

**Comment by buybackoff** [comment]
Even if the old tests are not the most beautiful thing in the world, they helped to catch many even subtly lock-free bugs.

**Comment by buybackoff** [comment]
Closed for #8
